lib/OperatingSystems/DefaultOs.php

```diff
@@ -141,50 +146,55 @@ public function getNetworkInfo(): array {
 	}
 
 	public function getNetworkInterfaces(): array {
-		$interfaces = glob('/sys/class/net/*') ?: [];
-		$result = [];
+		$data = [];
 
-		foreach ($interfaces as $interface) {
-			$iface = [];
-			$iface['interface'] = basename($interface);
-			$iface['mac'] = shell_exec('ip addr show dev ' . $iface['interface'] . ' | grep "link/ether " | cut -d \' \' -f 6  | cut -f 1 -d \'/\'');
-			$iface['ipv4'] = shell_exec('ip addr show dev ' . $iface['interface'] . ' | grep "inet " | cut -d \' \' -f 6  | cut -f 1 -d \'/\'');
-			$iface['ipv6'] = shell_exec('ip -o -6 addr show ' . $iface['interface'] . ' | sed -e \'s/^.*inet6 \([^ ]\+\).*/\1/\'');
-			if ($iface['interface'] !== 'lo') {
-				$iface['status'] = shell_exec('cat /sys/class/net/' . $iface['interface'] . '/operstate');
-				$iface['speed'] = (int)shell_exec('cat /sys/class/net/' . $iface['interface'] . '/speed');
-				if (isset($iface['speed']) && $iface['speed'] > 0) {
-					if ($iface['speed'] >= 1000) {
-						$iface['speed'] = $iface['speed'] / 1000 . ' Gbps';
-					} else {
-						$iface['speed'] = $iface['speed'] . ' Mbps';
-					}
-				} else {
-					$iface['speed'] = 'unknown';
+		foreach ($this->getNetInterfaces() as $interfaceName => $interface) {
+			$netInterface = new NetInterface();
+			$data[] = $netInterface;
+
+			$netInterface->setName($interfaceName);
+			$netInterface->setUp($interface['up']);
```

Comment:
Add name and up as constructor parameters maybe?

Reply:
Like the idea, thanks!